packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,225 @@
+import { Tag } from '../cloudformation/tag';
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * Properties Tags is a dictionary of tags as strings
+ */
+export interface Tags {
+    [key: string]: string
+}
+
+/**
+ * A an object of tags with value and properties
+ *
+ * This is used internally but not exported
+ */
+interface FullTags {
+    [key: string]: {value: string, props?: TagProps};
+}
+
+/**
+ * Properties for a tag
+ */
+export interface TagProps {
+    /**
+     * If true all child taggable `Constructs` will receive this tag
+     *
+     * @default true
+     */
+    propagate?: boolean;
+
+    /**
+     * If set propagated tags from parents will overwrite tags in children
+     *
+     * @default false
+     */
+    propagateOverride?: boolean;
+
+    /**
+     * If set this tag will overwrite existing tags
+     *
+     * @default true
+     */
+    overwrite?: boolean;
+}
+
+/**
+ * Properties for removing tags
+ */
+export interface RemoveProps {
+    /**
+     * If true prevent this tag form being set via propagation
+     *
+     * @default true
+     */
+    blockPropagate?: boolean;
+}
+
+/**
+ * TagManager facilitates a common implementation of tagging for Constructs.
+ *
+ * Each construct that wants to support tags should implement the `ITaggable`
+ * interface and properly pass tags to the `Resources` (Cloudformation) elements
+ * the `Construct` creates using `toCloudformation()` for lazy evaluations
+ */
+export class TagManager {
```

Comment:
I feel this class wants to be named `Tags`, feels cleaner to me. I understand there is a conflict with the current type `Tags` which also naturally is to be called that, but that could also be named `TagDict` or `TagValues` for example.

Reply:
I am on the fence here. It does do more than just hold tags, but logically it was the first name in mind until I hit the conflicts. I can start renaming if we have consensus, but I don't really want to plumb this change to find a stronger opinion reverting it. :)

Reply:
You are right, I have a feeling my opinion will be the minority one :). Don't change it.